### PR TITLE
jcastanos-patch-fwdict

### PR DESCRIPTION
**Pull request by jcastanos2**
get the fw dictionary form

### CODE REVIEW

File: roles/bmc_fw_update/tasks/main.yml [hunk 1]
```diff
@@ -83,11 +83,15 @@
 
 - name: Print BMC Version
   ansible.builtin.debug:
-    msg: "{{ get_bmc_facts_bmc_firmware }}"
+    msg: "{{ get_bmc_facts_all_fw_versions }}"
+
+- name: Store fw version we installed
+  ansible.builtin.set_fact:
+    got_fw_version: "{{ vars.get_bmc_facts_all_fw_versions['bmc_firmware'] }}"
```
Comment (glimchb) [inline roles/bmc_fw_update/tasks/main.yml:90]: I think linter should ask to have role name as variable prefix, for example `bmc_fw_update_got_fw_version`, no ?

**Comment by glimchb** [comment]
let's also try to remove linter exception from https://github.com/opiproject/ansible-opi-dpu/blob/main/.ansible-lint-ignore#L1

File: roles/bmc_fw_update/tasks/main.yml (hunk 1 above)
Comment (glimchb) [reply roles/bmc_fw_update/tasks/main.yml:90]: in a separate PR we should make `'bmc_firmware'` an argument

**Comment by glimchb** [comment]
I see checker for DCO failed, see https://github.com/apps/dco

need to use `-s` optino when you commit: 
```
git commit -s -m 'This is my commit message'
```
please fix existing commit message using:
```
git commit --signoff --amend --no-edit 
```

File: roles/get_bmc_facts/tasks/main.yml [hunk 2]
```diff
@@ -16,5 +16,6 @@
 
 - name: Extract all FW versions to facts
   ansible.builtin.set_fact:
-    "{{ 'get_bmc_facts_' + item['Id'] | lower }}": "{{ item['Version'] }}"
+    get_bmc_facts_all_fw_versions: "{{ get_bmc_facts_all_fw_versions | default({}) | combine({ item['Id'] | lower : item['Version'] }) }}"
   loop: "{{ result.redfish_facts.firmware.entries }}"
+
```
Comment (glimchb) [inline roles/get_bmc_facts/tasks/main.yml:21]: see lint fails on extra blank line here

File: roles/get_bmc_facts/tasks/main.yml [hunk 3]
```diff
@@ -16,5 +16,6 @@
 
 - name: Extract all FW versions to facts
   ansible.builtin.set_fact:
-    "{{ 'get_bmc_facts_' + item['Id'] | lower }}": "{{ item['Version'] }}"
+    get_bmc_facts_all_fw_versions: "{{ get_bmc_facts_all_fw_versions | default({}) | combine({ item['Id'] | lower : item['Version'] }) }}"
```
Comment (glimchb) [inline roles/get_bmc_facts/tasks/main.yml:19]: see linter suggestion below:
```
Jinja2 spacing could be improved: {{ get_bmc_facts_all_fw_versions | default({}) | combine({ item['Id'] | lower : item['Version'] }) }} -> {{ get_bmc_facts_all_fw_versions | default({}) | combine({item['Id'] | lower: item['Version']}) }}
```

File: roles/bmc_fw_update/tasks/main.yml (hunk 1 above)
Comment (jcastanos2) [reply roles/bmc_fw_update/tasks/main.yml:90]: Yes, it will be on the next one, to support cec and bmc (they are different versions)

**Review by glimchb — APPROVED** [review]

**Comment by glimchb** [comment]
example from test run on Nvidia BF-3 card:
```
2024-03-20 20:44:57 TASK [bmc_fw_update : Print BMC Version] ***************************************
2024-03-20 20:44:57 task path: /home/opiproject/ansible-opi-dpu/roles/bmc_fw_update/tasks/main.yml:12
2024-03-20 20:44:57 ok: [10.10.10.61] => {
2024-03-20 20:44:57     "msg": {
2024-03-20 20:44:57         "bluefield_fw_erot": "00.02.0127.0000_n02",
2024-03-20 20:44:57         "bmc_firmware": "BF-24.01-5",
2024-03-20 20:44:57         "dpu_atf": "v2.2(release):4.2.2-3-ged2a420",
2024-03-20 20:44:57         "dpu_board": "MT_0000001075",
2024-03-20 20:44:57         "dpu_bsp": "4.2.2.12958",
2024-03-20 20:44:57         "dpu_nic": "32.37.1300",
2024-03-20 20:44:57         "dpu_node": "a088:c203:005f:6648",
2024-03-20 20:44:57         "dpu_ofed": "MLNX_OFED_LINUX-23.07-0.5.0.0",
2024-03-20 20:44:57         "dpu_os": "DOCA_2.2.1_BSP_4.2.2_Ubuntu_22.04-13.23-09.prod",
2024-03-20 20:44:57         "dpu_sys_image": "a088:c203:005f:6648",
2024-03-20 20:44:57         "dpu_uefi": "4.2.2-2-g1174066",
2024-03-20 20:44:57         "golden_image_arm": "no valid golden_image_arm!!!!",
2024-03-20 20:44:57         "golden_image_nic": "no valid golden_image_nic!!!!"
2024-03-20 20:44:57     }
2024-03-20 20:44:57 }
2024-03-20 20:44:57
```

**Pull request merged**